eq-author-api/db/index.js

```diff
@@ -1,4 +1,48 @@
-var environment = process.env.NODE_ENV || "development";
-var config = require("../config/knexfile.js")[environment];
+const config = require("../config/knexfile.js");
+const createLogger = require("../utils/createLogger");
+const pinoMiddleware = require("express-pino-logger");
+const pino = pinoMiddleware();
+const logger = createLogger(pino.logger);
 
-module.exports = require("knex")(config);
\ No newline at end of file
+const getConfig = async function(secretId) {
+  if (secretId && secretId !== "") {
+    const AWS = require("aws-sdk");
+    const client = new AWS.SecretsManager({ region: "eu-west-1" });
+
+    let secretPromise = client.getSecretValue({ SecretId: secretId }).promise();
+
+    try {
+      let secret = await secretPromise;
+      let dbCredentials = JSON.parse(secret.SecretString);
+      config.connection = {
+        host: dbCredentials.host,
+        port: dbCredentials.port,
+        user: dbCredentials.username,
+        password: dbCredentials.password,
+        database: dbCredentials.dbname
+      };
+      return config;
+    } catch (error) {
+      logger.error(error);
+      process.exit();
+    }
+  }
+  return config;
+};
+
+let connection;
+
+function getConnection() {
+  if (!connection) {
+    getConfig(process.env.DB_SECRET_ID).then(conf => {
+      connection = require("knex")(conf);
+    });
+  }
+  return connection;
+}
+
+getConnection();
+
+module.exports = {
+  getConnection
```

Comment:
Could we export the connection directly? And then no one needs to make a call it just happens.

We could export getConnection and have that run as part of the initial start up? So that the connection is always available.